src/systemctl/systemctl.c

```diff
@@ -1984,25 +1984,55 @@ static int get_default(int argc, char *argv[], void *userdata) {
         return 0;
 }
 
-static void dump_unit_file_changes(const UnitFileChange *changes, unsigned n_changes) {
+static void dump_unit_file_changes(int r, const char *verb, const UnitFileChange *changes, unsigned n_changes) {
         unsigned i;
+        bool logged = false;
 
         assert(changes || n_changes == 0);
 
         for (i = 0; i < n_changes; i++)
                 switch(changes[i].type) {
                 case UNIT_FILE_SYMLINK:
-                        log_info("Created symlink %s, pointing to %s.", changes[i].path, changes[i].source);
+                        if (!arg_quiet)
+                                log_info("Created symlink %s, pointing to %s.", changes[i].path, changes[i].source);
                         break;
                 case UNIT_FILE_UNLINK:
-                        log_info("Removed %s.", changes[i].path);
+                        if (!arg_quiet)
+                                log_info("Removed %s.", changes[i].path);
                         break;
                 case UNIT_FILE_IS_MASKED:
-                        log_info("Unit %s is masked, ignoring.", changes[i].path);
+                        if (!arg_quiet)
+                                log_info("Unit %s is masked, ignoring.", changes[i].path);
+                        break;
+                case -EEXIST:
+                        if (changes[i].source)
+                                log_error("Failed to %s unit, file %s already exists and is a symlink to %s.",
```

Comment:
I figure we should use log_error_errno() here, and pass the error number... doesn't matter much, since the errno info is not used when outputting to a the console, but if people set SYSTEMD_LOG_LEVEL=journal, then it could be useful if the error number is stored in the structured data...